hyperv: Fix CPU hotadd

weltling opened the pull request:
The actual issue is the missing cpuid bits for the CPU dynamic partitioning support. There are two points to follow up, however:

- Adding a test involving the reproduce way as per #1799, best way is to be figured out to avoid UI usage.
- CpuManager functionality seems currently to have a bug introduced in 55a3a38e140c32748ed49acdacf6d54e59933048 at least for Windows guest. The number of CPUs as passed in `boot=` doesn't currently match the factual number in the VM. This needs more investigation and perhaps a separate fix.

Closes #1799.

Signed-off-by: Anatol Belski <anbelski@linux.microsoft.com>

sboeuf reviewed (approved):
LGTM

rbradford reviewed (changes requested):
Commit message needs work

weltling commented:
I expanded on implementation details in the commit message and rebased to the latest. Also added #2437 and #2438 to track the subsequent steps.

Thanks
